diff --git a/hw/vmapple/vmapple.c b/hw/vmapple/vmapple.c
--- a/hw/vmapple/vmapple.c
+++ b/hw/vmapple/vmapple.c
@@ -498,3 +498,4 @@ static void mach_vmapple_init(MachineState *machine)
         if (object_property_find(cpu, "has_el2")) {
-            object_property_set_bool(cpu, "has_el2", false, &error_fatal);
+            object_property_set_bool(cpu, "has_el2", vms->research_headless,
+                                     &error_fatal);
         }
